File: P5 2024-2/Paradigmas de Programacao/09-06/Python/Pessoa.py
class Pessoa:
    def __init__(self, nome, idade):
        self.nome = nome
        self.idade = idade
        self.endereco = None

    def adicionar_endereco(self, endereco):
        self.endereco = endereco

    def mostrar_informacoes(self):
        print(f"Nome: {self.nome}, Idade: {self.idade}")
        if self.endereco:
            print("Endereço: ")
            self.endereco.mostrar_endereco()
        else: 
            print("Endereço não disponível")

class Endereco:
    def __init__(self, rua, cidade, estado, cep):
        self.rua = rua
        self.cidade = cidade
        self.estado = estado
        self.cep = cep

    def mostrar_endereco(self):
        print(f"Rua: {self.rua}, Cidade: {self.cidade}, Estado: {self.estado}, CEP {self.cep}")

class Empresa:
    def __init__(self, nome, cnpj):
        self.nome = nome
        self.cnpj = cnpj
        self.funcionarios = []

    def contratar_funcionario(self, funcionario):
        self.funcionarios.append(funcionario)

    def listar_funcionarios(self):
        print(f"Funcionários da empresa: {self.nome}")
        for funcionario in self.funcionarios:
            print(funcionario.nome)

File: P5 2024-2/Paradigmas de Programacao/09-06/Golang/Pessoa.go
package main

import "fmt"

// Endereco struct represents an address.
type Endereco struct {
	Rua    string
	Cidade string
	Estado string
	Cep    string
}

func (e *Endereco) MostrarEndereco() {
	fmt.Printf("Rua: %s, Cidade: %s, Estado: %s, CEP: %s\n", e.Rua, e.Cidade, e.Estado, e.Cep)
}

type Pessoa struct {
	Nome     string
	Idade    int
	Endereco *Endereco
}

func (p *Pessoa) AdicionarEndereco(endereco *Endereco) {
	p.Endereco = endereco
}

func (p *Pessoa) MostrarInformacoes() {
	fmt.Printf("Nome: %s, Idade: %d\n", p.Nome, p.Idade)
	if p.Endereco != nil {
		fmt.Println("Endereço: ")
		p.Endereco.MostrarEndereco()
	} else {
		fmt.Println("Endereço não disponível")
	}
}

type Empresa struct {
	Nome         string
	Cnpj         string
	Funcionarios []*Pessoa
}

func (e *Empresa) ContratarFuncionario(funcionario *Pessoa) {
	e.Funcionarios = append(e.Funcionarios, funcionario)
}

func (e *Empresa) ListarFuncionarios() {
	fmt.Printf("Funcionários da empresa: %s\n", e.Nome)
	for _, funcionario := range e.Funcionarios {
		fmt.Println(funcionario.Nome)
	}
}
